### ISSUE TITLE
services.locate.enable = true; throws error in rebuild!

### ISSUE THREAD

**Issue by ktaf** [NixOS/nixpkgs#258984]
### Describe the bug
After this commit 524181b, while I have services.locate.enable = true; in my configuration.nix, rebuilding NixOS throws errors!

Most searchable one:

error: attribute 'override' missing
environment.systemPackages = [ (cfg.locate.override { dbfile = cfg.output; }) ];

### Steps To Reproduce
1. Add services.locate.enable = true; in configuration.nix
2. sudo nixos-rebuild switch --flake .#USER

### Expected behavior
Rebuild pkgs...
### Screenshots
![Screenshot-2023-10-04T14:40:35](https://github.com/NixOS/nixpkgs/assets/16401735/3fd1d70d-434c-459a-b52b-56527338a75f)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.5, NixOS, 23.11 (Tapir), 23.11.20231001.fdd898f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(kourosh): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos````

**Comment by bobvanderlinden** [comment]
I ran into the same issue on `nixos-unstable`. I moved to `nixpkgs-unstable` for now, where the problem is resolved. I'm not quite sure why it was failing on `nixos-unstable` yet.

**Comment by eclairevoyant** [comment]
should be fixed by #258575
